File: Submissions/#0577_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int m = sc.nextInt();

    int[][] count = new int[m][5];

    for (int i = 0; i < n; i++) {
      String S = sc.next();
      char[] s = S.toCharArray();
      for (int j = 0; j < m; j++) {
        count[j][(int) s[j] - 'A']++;
      }
    }
  
    int answer = 0;
    int[] mx = new int[m];

    for (int i = 0; i < m; i++) {
      int p = sc.nextInt();
      for (int j = 0; j < 5; j++) {
        mx[i] = Math.max(mx[i], count[i][j]);
      }
      answer += mx[i] * p;
    }

    sc.close();

    System.out.println(answer);

  }

}

File: Submissions/#0577_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
 
  int n, m;
  cin >> n >> m;

  vector<vector<int>> count(m,vector<int>(5));
  
  for (int i = 0; i < n; i++) {
    string s;
    cin >> s;
    for (int j = 0; j < m; j++) {
      count[j][(int) s[j] - 'A']++;
    }
  }

  int answer = 0;
  vector<int> mx(m);

  for (int i = 0; i < m; i++) {
    int p;
    cin >> p;
    for (int j = 0; j < 5; j++) {
      mx[i] = max(mx[i], count[i][j]);
    }
    answer += mx[i] * p;
  }

  cout << answer << endl;

}